adapter/src/dbt/adapters/fal_experimental/support/trino.py

```diff
@@ -0,0 +1,30 @@
+from typing import Any, Dict, Optional
+from dbt.adapters.base import BaseAdapter, Credentials
+from trino.sqlalchemy import URL
+from dbt.adapters.trino.connections import TrinoCredentials, TrinoNoneCredentials
+
+def create_engine(adapter: BaseAdapter) -> Any:
+    import sqlalchemy
+    creds = adapter.config.credentials._db_creds
+
+    # Can be None
+    connect_args = _build_connect_args(creds)
+
+    url = URL(
+        host=creds.host,
+        port=creds.port,
+        catalog=creds.database,
+        user=creds.user
+    )
+    return sqlalchemy.create_engine(url, connect_args=connect_args)
+
+def _build_connect_args(credentials: TrinoCredentials) -> Optional[Dict[str, Any]]:
+    if credentials is TrinoNoneCredentials:
```

Comment:
Should this be an isinstace check?
```
isinstance(credentials, TrinoNoneCredentials)
```

Reply:
It should, so maybe the comment `# Can be None` is also not really correct.